### PR TITLE
SOC-1194 - Not Found page for Discussions

### PR DESCRIPTION
**Pull request by armonr**
Adds a new 404 page at `/404`

/CC @mklucsarits @garthwebb @bkoval 

https://wikia-inc.atlassian.net/browse/SOC-1194

### CODE REVIEW

**Comment by rafalkalinski** [comment]
Correct me if I'm wrong, but I don't think, we should redirect user to the /404 if the url addres is 'Not Found'. It would be good to stay at the same address user wanted to visit and display the message 'Not found' there.
@kvas-damian @garthwebb @bkoval ?

**Comment by kvas-damian** [comment]
I agree with @rafalkalinski, additionally ```/404``` returns status 200 which is against SEO.

File: front/styles/main/layout/_page-wrapper.scss [hunk 1]
```diff
@@ -25,6 +24,12 @@
...
 }
 
-.site-head.theme-bar ~ .page-wrapper {
-	margin-top: 60px;
+@media #{$desktop-range} {
+	.page-wrapper {
+		margin-top: 50px;
+	}
+
+	.site-head.theme-bar ~ .page-wrapper {
```
Comment (bkoval) [inline front/styles/main/layout/_page-wrapper.scss]: Out of curiosity: why do we need to check for page-wrapper as a sibling of theme-bar? I guess there's a good reason for it :)

File: front/styles/main/module/_discussion.scss [hunk 2]
```diff
@@ -76,6 +76,26 @@ $horizontal-padding: 15px;
 		}
 	}
 
+	.error-content {
+		background-color: $color-white;
+		border-radius: 8px;
+		text-align: center;
+
+		.error-message {
+			color: $aluminum;
```
Comment (bkoval) [inline front/styles/main/module/_discussion.scss]: Is this a foundation variable? If yes, we probably should avoid calling them directly as there are plans to evaluate dropping foundation in styleguide.

File: front/styles/main/module/_discussion.scss [hunk 3]
```diff
@@ -76,6 +76,26 @@ $horizontal-padding: 15px;
...
+			font-size: rem-calc(14);
+		}
+
+		.error-action {
+			font-size: rem-calc(12);
+			margin-top: 8px;
+
+			a {
+				color: $oil;
```
Comment (bkoval) [inline front/styles/main/module/_discussion.scss]: Same here :)

File: front/templates/main/discussion/notFound.hbs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<div class="minimized discussion-hero-unit"></div>
+<div class="discussion-page row">
+	<div class="rail large-2 columns">
+		{{community-badge}}
+		{{discussion-back-button}}
+	</div>
+	<hr class="theme desktop-hidden">
+	<div class="discussion post small-12 large-8 columns">
+		<div class="error-container">
+			<div class="error-content">
```
Comment (bkoval) [inline front/templates/main/discussion/notFound.hbs]: Wow, is there any way to simplify the markup a bit? There are 3 layers of divs containing just another div inside :)

**Comment by bkoval** [comment]
@armonr Are there a lot of changes to i18n here? It's not visible at the first sight as the diff is pretty big, maybe because of changed whitespacing?

**Comment by rogatty** [comment]
@bkoval after merging the latest dev it should stop being a problem.

**Comment by bkoval** [comment]
@rogatty thanks!

@rafalkalinski Yeah, that feels more suitable to me as well.

File: front/styles/main/layout/_page-wrapper.scss (hunk 1 above)
Comment (armonr) [inline front/styles/main/layout/_page-wrapper.scss]: I just moved this rule here, but maybe @mklucsarits can shed light on it. https://github.com/Wikia/mercury/commit/fc4b7f53df8f7230ce524aaf0ae329b9fb336d06#diff-85a9c48ab225cb8722f97bdcea440979R30

Comment (mklucsarits) [inline front/styles/main/layout/_page-wrapper.scss]: I honestly don't even remember. And with the changes removing page-wrapper usage on discussion pages, and changing implementation of the theme bar after top-bar integration, this style might not be used anymore.

**Comment by garthwebb** [comment]
@rafalkalinski, @armonr, re: redirecting to the 404 page.  I agree with @rafalkalinski; if you've mistyped a URL, its really annoying to have to type in the whole URL again rather than correct the mistake.

**Comment by armonr** [comment]
@garthwebb & @rafalkalinski: Yes, agreed. I have been working on that; ember transitioning is proving very tricky to me in this case; but stay tuned for the fix!

**Comment by rafalkalinski** [comment]
The story was done as a part of SOC-1193

**Pull request closed without merging**